### PR TITLE
Fix PagesIndex.getEstimatedSize for empty index

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

**Comment by nezihyigitbasi** [comment]
What kind of problem was this causing?

**Comment by findepi** [comment]
I wouldn't say any. Just for consistency

**Comment by nezihyigitbasi** [comment]
It would be awesome if you can also add some tests to verify PagesIndex estimated size as I don't think we have any tests for that.

**Comment by findepi** [comment]
@nezihyigitbasi, a test added.

File: presto-main/src/test/java/com/facebook/presto/operator/TestPagesIndex.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        pagesIndex.addPage(somePage(types));
+        long estimatedSizeWithOnePage = pagesIndex.getEstimatedSize().toBytes();
+        assertTrue(estimatedSizeWithOnePage > initialEstimatedSize, "Estimated size should grow after adding a page");
+
+        pagesIndex.addPage(somePage(types));
+        long estimatedSizeWithTwoPages = pagesIndex.getEstimatedSize().toBytes();
+        assertEquals(
+                estimatedSizeWithTwoPages,
+                initialEstimatedSize + (estimatedSizeWithOnePage - initialEstimatedSize) * 2,
+                "Estimated size should grow linearly as long as we don't pass expectedPositions");
```
Comment (nezihyigitbasi) [inline presto-main/src/test/java/com/facebook/presto/operator/TestPagesIndex.java]: Can you also add a test for `compact()`? We can assert that the estimated size of compacted pages index should be `<=` the original size.

File: presto-main/src/test/java/com/facebook/presto/operator/TestPagesIndex.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        assertTrue(initialEstimatedSize.toBytes() > 0, "No initial estimated size");
+    }
+
+    @Test
+    public void testEstimatedSize()
+    {
+        List<Type> types = ImmutableList.of(BIGINT, VARCHAR);
+
+        PagesIndex pagesIndex = newPagesIndex(types, 30);
+        long initialEstimatedSize = pagesIndex.getEstimatedSize().toBytes();
```
Comment (nezihyigitbasi) [inline presto-main/src/test/java/com/facebook/presto/operator/TestPagesIndex.java]: You can put the `assertTrue` in the above `testInitialEstimatedSize()` here and remove that method.

File: presto-main/src/test/java/com/facebook/presto/operator/TestPagesIndex.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        long estimatedSizeWithTwoPages = pagesIndex.getEstimatedSize().toBytes();
+        assertEquals(
+                estimatedSizeWithTwoPages,
+                initialEstimatedSize + (estimatedSizeWithOnePage - initialEstimatedSize) * 2,
+                "Estimated size should grow linearly as long as we don't pass expectedPositions");
+    }
+
+    private static PagesIndex newPagesIndex(List<Type> types, int expectedPositions)
+    {
+        return new PagesIndex.TestingFactory().newPagesIndex(types, expectedPositions);
```
Comment (findepi) [inline presto-main/src/test/java/com/facebook/presto/operator/TestPagesIndex.java]: All other tests use `PagesIndex.TestingFactory` as here. I would import if this was `TestingPagesIndexFactory` or something else more meaningful.

File: presto-main/src/test/java/com/facebook/presto/operator/TestPagesIndex.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    private static Page somePage(List<Type> types)
```
Comment (findepi) [inline presto-main/src/test/java/com/facebook/presto/operator/TestPagesIndex.java]: I wrote "some" to indicate the page contains something but caller doesn't control what. I would expect `newPage(List<Type>)` to produce an empty page.

**Comment by findepi** [comment]
@nezihyigitbasi , squashed and rebased. Travis happy.

**Comment by nezihyigitbasi** [comment]
Merged, thanks!

**Pull request closed without merging**